.github/workflows/check.yml

```diff
@@ -0,0 +1,106 @@
+name: "Check Workflow"
+
+on: [push, pull_request]
+
+jobs:
+
+  perl-versions:
+    runs-on: ubuntu-latest
+    name: 'perl-versions default'
+    outputs:
+      perl-versions: ${{ steps.action.outputs.perl-versions }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "uses perl-versions"
+        id: action
+        uses: ./
+
+  since-v520:
+    runs-on: ubuntu-latest
+    name: 'since v5.20'
+    outputs:
+      perl-versions: ${{ steps.action.outputs.perl-versions }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "uses perl-versions"
+        id: action
+        uses: ./
+        with:
+          since-perl: v5.20
+
+  since-520:
+    runs-on: ubuntu-latest
+    name: 'since 5.20'
+    outputs:
+      perl-versions: ${{ steps.action.outputs.perl-versions }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "uses perl-versions"
+        id: action
+        uses: ./
+        with:
+          since-perl: "5.20"
+
+  since-536-with-devel:
+    runs-on: ubuntu-latest
+    name: 'since 5.36 with devel'
+    outputs:
+      perl-versions: ${{ steps.action.outputs.perl-versions }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "uses perl-versions"
+        id: action
+        uses: ./
+        with:
+          since-perl: 5.36
+          with-devel: true
+
+  test-matrix:
+    runs-on: ubuntu-latest
+    needs:
+      - since-536-with-devel
+    strategy:
+      fail-fast: false
+      matrix:
+        perl-versions: ${{ fromJson (needs.since-536-with-devel.outputs.perl-versions) }}
+    steps:
+      - run: echo "Hello"
+
+  perl-tester:
+    runs-on: ubuntu-latest
+    needs:
+      - since-536-with-devel
+    name: "Perl ${{ matrix.perl-version }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        perl-version: ${{ fromJson (needs.since-536-with-devel.outputs.perl-versions) }}
+    container:
+      image: perldocker/perl-tester:${{ matrix.perl-version }}
```

Comment:
fyi, perl-versions are used to publish perl-tester docker images, so upcoming v5.40 will be first added here and perl-tester:5.40 will exist later

Reply:
yeap I was expecting such addition in a few days, but looks like we control the list manually and at the same time we are adding it we should now also add it to the poor man test I ve added there, should be ok I think?

Reply:
hm, let's see, we can fix issues later
workflow I have growing in my mind is:
- test version validity using authority list (probably reusing `perlbrew available`)
- trigger perl-tester publish workflow when new version is added

Reply:
definitively good improvements we can add in a short future! 
I'm trying to get that pipeline ready for release so we can start using it in multiple distro 

Let me know if you are fine with the current state before merging
thanks